Allow the Spice Labs CLI to Scan Itself

CJ-cmd-byte opened the pull request:
PR for: https://github.com/spice-labs-inc/internal-docs/issues/308 `Spice Labs CLI`

dpletter commented on .github/workflows/publish.yml:
```diff
@@ -134,37 +150,54 @@ jobs:
-      - name: Build and publish all jars (regular + fat)
+      - name: Build and publish all jars (Maven Central)
         id: build-jars-maven-central
+        if: startsWith(github.ref, 'refs/tags/v') && steps.onmain.outputs.on_main == 'true'
```
let's not change this step or its comments

dpletter commented on .github/workflows/publish.yml:
```diff
@@ -134,37 +150,54 @@ jobs:
+            --tag "${{ github.event.repository.name }}"
 
       - name: Upload wrapper scripts to GitHub Release
+        if: startsWith(github.ref, 'refs/tags/v') && steps.onmain.outputs.on_main == 'true'
```
seems like these conditionals and the additional branch trigger were here for testing, but can we remove them now?

CJ-cmd-byte commented:
@dpletter - hopefully this is a minimal change that addresses all of your comments and concerns

- removed temporary `mj/self-scan` branch from the trigger 
- removed the “pull image → save tar → rescan” sequence so we only scan the built JARs in target

dpletter commented on .github/workflows/publish.yml:
```diff
@@ -1,64 +1,83 @@
 
       - name: Validate and extract version from tag
         id: validate
+        if: startsWith(github.ref, 'refs/tags/v')
```
why was this necessary?

dpletter commented on .github/workflows/publish.yml:
```diff
@@ -134,37 +149,54 @@ jobs:
+          GPG_PASSPHRASE: "${{ secrets.MAVEN_CENTRAL_GPG_PASSPHRASE }}"
 
-      - name: Build and publish all jars (regular + fat)
+      - name: Build and publish all jars (Maven Central)
```
the existing comment was more correct and meaningful

dpletter commented on .github/workflows/publish.yml:
```diff
@@ -173,10 +205,10 @@ jobs:
   docker_image:
     name: Build and Push Docker Image
-    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/v')
```
not needed

dpletter commented on .github/workflows/publish.yml:
```diff
@@ -196,97 +228,88 @@ jobs:
-
-      - name: Check if release is a pre-release
-        id: prerelease
+      - name: Ensure tag commit is on main
```
removing the download of the fat jar makes me nervous. i don't understand it.  this was done so that we are clear that we are using the correct jar in the image

dpletter commented on .github/workflows/publish.yml:
```diff
@@ -196,97 +228,88 @@ jobs:
-          images: spicelabs/spice-labs-cli
+          images: "${{ env.IMAGE_NAME }}"
           tags: |
             type=semver,pattern={{version}}
```
why are we removing tags?

dpletter commented on .github/workflows/publish.yml:
```diff
@@ -196,97 +228,88 @@ jobs:
+            MESSAGE="Deployment result:"
             if [ "$JAR_STATUS" != "success" ]; then
-              MESSAGE="$MESSAGE ❌ JAR publishing failed."
+              MESSAGE="$MESSAGE JAR publishing failed."
```
why did the format need to change? i worked hard to line this format up with all of our other builds

CJ-cmd-byte commented:
This was a mistake on my part, I’ll correct it.

Manjula-Jayne Kumar

Spice Labs, Inc.  |  Systems Engineer

+64 204 013 7666


On Thu, 18 Sep 2025 at 09:54, Dani Pletter ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In .github/workflows/publish.yml
> <https://github.com/spice-labs-inc/spice-labs-cli/pull/106#discussion_r2356871101>
> :
>
> >              if [ "$JAR_STATUS" != "success" ]; then
> -              MESSAGE="$MESSAGE ❌ JAR publishing failed."
> +              MESSAGE="$MESSAGE JAR publishing failed."
>
> why did the format need to change? i worked hard to line this format up
> with all of our other builds
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/spice-labs-inc/spice-labs-cli/pull/106#pullrequestreview-3236387138>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BNEBUW6FN3AKYKQOP6YJHGT3THKCTAVCNFSM6AAAAACF7YVI56VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTEMZWGM4DOMJTHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
